src/credentials/attestation_verifier/JSONFileRevocationSet.cpp

```diff
@@ -0,0 +1,189 @@
+                                                     sizeof(crlIssuerSubjectKeyIdBuf)) == Crypto::kAuthorityKeyIdentifierLength,
+                                AttestationVerificationResult::kInvalidArgument);
+
+            size_t crlSignerCertificateMaxLength = BASE64_MAX_DECODED_LEN(jsonCrlIssuerName.asString().size());
+
+            Platform::ScopedMemoryBuffer<uint8_t> crlSignerCertificate;
+
+            ReturnErrorCodeIf(!crlSignerCertificate.Alloc(crlSignerCertificateMaxLength), AttestationVerificationResult::kNoMemory);
+
+            // 2.
+            size_t crlSignerCertificateLength =
+                Base64Decode(jsonCrlIssuerName.asString().c_str(), jsonCrlIssuerName.asString().size(), crlSignerCertificate.Get());
+            VerifyOrReturnError(crlSignerCertificateLength > 0 && crlSignerCertificateLength != UINT16_MAX,
+                                AttestationVerificationResult::kInternalError);
+
+            // 3.
+            if (isPaa)
+            {
+                Crypto::AttestationCertVidPid vid;
+
+                Crypto::ExtractVIDPIDFromAttributeString(
+                    Crypto::DNAttrType::kCommonName, ByteSpan(crlSignerCertificate.Get(), crlSignerCertificateLength), vid, vid);
+
+                if (vid.mVendorId.HasValue())
+                {
+                    if (vid.mVendorId.Value() != vidPidUnderTest.mVendorId.Value())
+                    {
+                        // VID does not match. Stop further processing and continue to next entry.
+                        continue;
+                    }
+                }
+            }
+            // 4.
+            else
+            {
+                Crypto::AttestationCertVidPid vidPid;
+
+                Crypto::ExtractVIDPIDFromAttributeString(Crypto::DNAttrType::kCommonName,
+                                                         ByteSpan(crlSignerCertificate.Get(), crlSignerCertificateLength), vidPid,
+                                                         vidPid);
+
+                if (vidPid.mVendorId.HasValue() && vidPid.mVendorId.Value() != vidPidUnderTest.mVendorId.Value())
+                {
+                    // VID does not match. Stop further processing and continue to next entry.
+                    continue;
+                }
+
+                if (vidPid.mProductId.HasValue())
+                {
+                    if (vidPid.mProductId.Value() != vidPidUnderTest.mProductId.Value())
+                    {
+                        // PID does not match. Stop further processing and continue to next entry.
+                        continue;
+                    }
+                }
+            }
+
+            // 7.a Perform CRLFile validation
+            if (authorityKeyId.data_equal(crlIssuerSubjectKeyId) == false)
+            {
+                continue;
+            }
+
+            // TODO: 7.b
+
+            // TODO: 8.
```

Comment:
Tracking issues, with links to them here, please.

Reply:
I'll add better description on a future commit